src/__tests__/GoalParser.test.ts

```diff
@@ -70,3 +77,32 @@ describe("getTime function", () => {
     expect(timeSuggestion()).toEqual(time());
   });
 });
+function urlDetection() {
+  if (goalUrl.title.search(urlDetector) !== -1) {
+    return {
+      urlData,
+      suggestion: { Link: "Link" },
+    };
+  }
+
+  return {
+    urlData,
+  };
+}
+describe("getUrl function", () => {
+  it("should return link : Link", () => {
+    function urlTest() {
+      if (goalUrl.title.search(urlDetector) !== -1) {
+        return {
+          urlData,
+          suggestion: { Link: "Link" },
+        };
+      }
+
+      return {
+        urlData,
+      };
+    }
+    expect(urlDetection()).toEqual(urlTest());
```

Comment:
The test is a bit confusing for me to understand. But when you are comparing the output it's better to have the hard-code value rather than a function call.